### ISSUE TITLE
Cross Validation/Data augmentation

### ISSUE THREAD

**Issue by samieti02** [Kamnitsask/deepmedic#132]
Hello dear Sir,

You wrote in your publication that you trained the network with a 5-fold cross validation. How can we achieve it in the code? I didnt see a parameter in the config files that help us to do it.

For the cross valiadation, how should we choose the fold?
Lets say for example we have the data p1, ..., p10
We can have the data sorting like this, and with the following folds
`p1 p2 | p3 p4 | p5 p6 | p7 p8 | p9 p10`

You could also have 
`p4 p2 | p3 p8 | p5 p7 | p10 p1 | p9 p6`
... Or another permutation

Does it make sense to calculate the error for all the permutations ?

Or simply work with the data sorting like this `p1 p2 | p3 p4 | p5 p6 | p7 p8 | p9 p10` like in the first case?
And is there a rule to adjust the hyperparameters after we calculated ther error of neural network 1.
For neural  network 2, how would you set the parameters(Learning rate, momentum, ...)? If you had for example learning rate = 1 and for that value, we have an error, how would you set the parameters of network 2, learning  rate= 1.5 or learning rate = 5? Is there a rule for that ?

Second, you also told that you use data augmentation - where transformations are used. Can we either plot the augmented data or save them as nifti data, for further investigation?

Thanks Sir,

**Comment by Kamnitsask** [comment]
Cross validation: It is not done in the code. You do it yourself by splitting the data accordingly.
Any "random" permutation that captures the data correctly is fine. (eg avoid taking the 1st 1/5th of the database, then the 2nd/5tth etc, because database can be ordered in some non-random way, so now folds wouldnt be random).
The two you suggest should be fine.

Augmentation: Done internally. Code cant save the images currently. You would need to alter it yourself. This is not a feature I am planning to do yet. You need to look into routines/training and sampling classes (check what training calls).

Regarding hyper-parameter search: This is far from a technical issue (which this place is for), and a huge discussion. This is based on experience and intuitition, plus some random search.

Note that issues are more helpful to future users and easier to answer if they only touch one subject. (I dont want to be rude, just trying my best to support the project, and managing issues is part of it ;-) )

I hope the above helps
K
